packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -113,103 +139,79 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
   }
 
+  /**
+   * Chain maybe started from a genesis state file but we may still need to store genesis deposits.
+   */
+  public async ensurePersistGenesisDeposits(eth1BlockHash: string): Promise<void> {
+    if (!this.opts.enabled) {
+      return;
+    }
+    const block = await this.getBlock(eth1BlockHash);
+    await this.processBlocksUntil(block.number);
+  }
+
   public async getLastProcessedBlockTag(): Promise<string | number> {
     const lastEth1Data = await this.db.eth1Data.lastValue();
     return lastEth1Data ? toHexString(lastEth1Data.blockHash) : this.opts.depositContract.deployedAt;
   }
-  public async getLastProcessedDepositIndex(): Promise<number> {
-    const lastStoredIndex = await this.db.depositDataRoot.lastKey();
-    return lastStoredIndex === null ? -1 : lastStoredIndex;
-  }
 
   public async onNewEth1Block(blockNumber: number): Promise<void> {
     const followBlockNumber = blockNumber - this.config.params.ETH1_FOLLOW_DISTANCE;
-    if (followBlockNumber > 0 && followBlockNumber > this.lastProcessedEth1BlockNumber) {
-      await this.processBlocks(followBlockNumber);
+    if (followBlockNumber > 0) {
+      await this.processBlocksUntil(followBlockNumber);
     }
   }
 
-  public async processBlocks(toNumber: number): Promise<void> {
-    let blockNumber = this.lastProcessedEth1BlockNumber + 1;
-    while (blockNumber <= toNumber && await this.processBlock(blockNumber)) {
-      blockNumber++;
+  public async processBlocksUntil(blockNumber: number): Promise<void> {
+    if (blockNumber > this.lastProcessedEth1BlockNumber) {
+      await this.getAndProcessDepositEvents(this.lastProcessedEth1BlockNumber + 1, blockNumber);
     }
   }
 
   /**
-   * Process an eth1 block for DepositEvents and Eth1Data
-   *
-   * Must process blocks in order with no gaps
-   *
-   * Returns true if processing was successful
+   * Process blocks in bulk.
+   * Must process blocks in order with no gaps.
+   * @param fromBlockNbr
+   * @param toBlockNbr
    */
-  public async processBlock(blockNumber: number): Promise<boolean> {
-    if (!this.started) {
+  public async getAndProcessDepositEvents(fromBlockNbr: number, toBlockNbr: number): Promise<void> {
+    if (!this.started && !this.isScanEth1ForGenesis) {
       this.logger.verbose("Eth1 notifier must be started to process a block");
-      return false;
+      return;
+    }
+    if (toBlockNbr < fromBlockNbr) {
+      return;
     }
     this.processingBlock = true;
-    this.logger.verbose(`Processing eth1 block ${blockNumber}`);
-    if (blockNumber !== this.lastProcessedEth1BlockNumber + 1) {
-      this.logger.verbose(
-        `eth1 block out of order. expected: ${this.lastProcessedEth1BlockNumber + 1} actual: ${blockNumber}`
+    this.logger.verbose(`Processing eth1 block ${fromBlockNbr} to ${toBlockNbr}`);
+    try {
+      const depositEvents  = await this.getDepositEvents(fromBlockNbr, toBlockNbr);
+      this.logger.info(`Found ${depositEvents.length} deposit events`);
```

Comment:
maybe add block number in which deposits are found

Reply:
this is deposit events of a block range, logged the block range in above line. We can also log deposit events of a single block inside `processDepositEvents`

Reply:
added that